processing/src/main/java/io/druid/segment/IndexMaker.java

```diff
@@ -473,6 +494,10 @@ private static File makeIndexFiles(
     progress.start();
     progress.progress();
 
+    if (metricAggs != null) {
+      segmentMetadata = SegmentMetadataHelper.withAggregators(segmentMetadata, metricAggs);
```

Comment:
Usually I prefer making a new variable for stuff like this (segmentMetadataWithAggregators?) rather than reassigning to a method parameter. Mostly that's because I usually find it harder to understand what's going on when method parameters get reassigned.

Not sure what other people's thoughts on this is though, as it is a matter of style…